packages/swapper/src/swappers/RelaySwapper/endpoints.ts

```diff
@@ -0,0 +1,223 @@
+
+    const relayQuote = tradeQuoteMetadata.get(tradeQuote.id)
+
+    if (!relayQuote) throw Error(`missing trade quote metadata for quoteId ${tradeQuote.id}`)
+
+    const swapSteps = relayQuote.steps.filter(step => step.id !== 'approve')
+    const { to, value, data } = swapSteps[stepIndex].items?.[0].data ?? {}
+
+    const { networkFeeCryptoBaseUnit } = await evm.getFees({
+      adapter: assertGetEvmChainAdapter(chainId),
+      data: data.toString(),
+      to,
+      // This looks odd but we need this, else unchained estimate calls will fail with:
+      // "invalid decimal value (argument=\"value\", value=\"0x0\", code=INVALID_ARGUMENT, version=bignumber/5.7.0)"
+      value: bnOrZero(value.toString()).toString(),
+      from,
+      supportsEIP1559,
+    })
+
+    return networkFeeCryptoBaseUnit
+  },
+
+  checkTradeStatus: async ({
+    quoteId,
+    txHash,
+    stepIndex,
+    chainId,
+    accountId,
+    config,
+    fetchIsSmartContractAddressQuery,
+    assertGetEvmChainAdapter,
+  }): Promise<{
+    status: TxStatus
+    buyTxHash: string | undefined
+    message: string | [string, InterpolationOptions] | undefined
+  }> => {
+    const maybeSafeTransactionStatus = await checkSafeTransactionStatus({
+      txHash,
+      chainId,
+      assertGetEvmChainAdapter,
+      accountId,
+      fetchIsSmartContractAddressQuery,
+    })
+
+    if (maybeSafeTransactionStatus) {
+      // return any safe transaction status that has not yet executed on chain (no buyTxHash)
+      if (!maybeSafeTransactionStatus.buyTxHash) return maybeSafeTransactionStatus
+
+      // The safe buyTxHash is the on chain transaction hash (not the safe transaction hash).
+      // Mutate txHash and continue with regular status check flow.
+      txHash = maybeSafeTransactionStatus.buyTxHash
+    }
+
+    const relayQuote = tradeQuoteMetadata.get(quoteId)
+
+    if (!relayQuote) throw Error(`missing trade quote metadata for quoteId ${quoteId}`)
+
+    const swapSteps = relayQuote.steps.filter(step => step.id !== 'approve')
+
+    const requestId = swapSteps[stepIndex].requestId
+
+    const maybeStatusResponse = await relayService.get<RelayStatus>(
+      `${config.VITE_RELAY_API_URL}intents/status/v2?requestId=${requestId}`,
+    )
+
+    if (maybeStatusResponse.isErr()) {
+      return {
+        buyTxHash: undefined,
+        status: TxStatus.Unknown,
+        message: undefined,
+      }
+    }
+
+    const { data: statusResponse } = maybeStatusResponse.unwrap()
+
+    const status = (() => {
+      switch (statusResponse.status) {
+        case 'success':
+          return TxStatus.Confirmed
+        case 'pending':
+          return TxStatus.Pending
+        case 'failed':
+          return TxStatus.Failed
+        default:
+          return TxStatus.Unknown
+      }
+    })()
+
+    return {
+      // We have an out Tx hash (either same or cross-chain) for this step, so we consider the Tx (effectively, the step) confirmed
```

Comment:
Comment doesn't apply here, we directly return `status` as-is, without leveraging out Tx for status detection

Reply:
Removed in [4483912](https://github.com/shapeshift/web/pull/9188/commits/448391212005a43def0ab43b38186507be95ff6c)